plugins/node/opentelemetry-instrumentation-mongodb/test/mongodb-v4.metrics.test.ts

```diff
@@ -0,0 +1,169 @@
+  exporter: inMemoryMetricsExporter,
+  exportIntervalMillis: 100,
+  exportTimeoutMillis: 100,
+});
+
+otelTestingMeterProvider.addMetricReader(metricReader);
+
+import { registerInstrumentationTesting } from '@opentelemetry/contrib-test-utils';
+const instrumentation = registerInstrumentationTesting(
+  new MongoDBInstrumentation()
+);
+
+instrumentation.setMeterProvider(otelTestingMeterProvider);
+
+import { accessCollection, DEFAULT_MONGO_HOST } from './utils';
+
+import * as assert from 'assert';
+
+async function waitForNumberOfExports(
+  exporter: InMemoryMetricExporter,
+  numberOfExports: number
+): Promise<ResourceMetrics[]> {
+  if (numberOfExports <= 0) {
+    throw new Error('numberOfExports must be greater than or equal to 0');
+  }
+
+  let totalExports = 0;
+  while (totalExports < numberOfExports) {
+    await new Promise(resolve => setTimeout(resolve, 20));
+    const exportedMetrics = exporter.getMetrics();
+    totalExports = exportedMetrics.length;
+  }
+
+  return exporter.getMetrics();
+}
+
+describe('MongoDBInstrumentation-Metrics', () => {
+  // For these tests, mongo must be running. Add RUN_MONGODB_TESTS to run
+  // these tests.
+  const RUN_MONGODB_TESTS = process.env.RUN_MONGODB_TESTS as string;
+  let shouldTest = true;
+  if (!RUN_MONGODB_TESTS) {
+    console.log('Skipping test-mongodb. Run MongoDB to test');
+    shouldTest = false;
+  }
+
+  const HOST = process.env.MONGODB_HOST || DEFAULT_MONGO_HOST;
+  const PORT = process.env.MONGODB_PORT || '27017';
+  const DB_NAME = process.env.MONGODB_DB || 'opentelemetry-tests-metrics';
+  const COLLECTION_NAME = 'test-metrics';
+  const URL = `mongodb://${HOST}:${PORT}/${DB_NAME}`;
+
+  let client: mongodb.MongoClient;
+  let collection: mongodb.Collection;
+  beforeEach(function mongoBeforeEach(done) {
+    // Skipping all tests in beforeEach() is a workaround. Mocha does not work
+    // properly when skipping tests in before() on nested describe() calls.
+    // https://github.com/mochajs/mocha/issues/2819
+    if (!shouldTest) {
+      this.skip();
+    }
+    inMemoryMetricsExporter.reset();
+    done();
+  });
+
+  it('Should add connection usage metrics', async () => {
+    const result = await accessCollection(URL, DB_NAME, COLLECTION_NAME);
+    client = result.client;
+    collection = result.collection;
+    const insertData = [{ a: 1 }, { a: 2 }, { a: 3 }];
+    await collection.insertMany(insertData);
+    await collection.deleteMany({});
+    let exportedMetrics = await waitForNumberOfExports(
+      inMemoryMetricsExporter,
+      1
+    );
+
+    assert.strictEqual(exportedMetrics.length, 1);
+    let metrics = exportedMetrics[0].scopeMetrics[0].metrics;
+    assert.strictEqual(metrics.length, 1);
+    assert.strictEqual(metrics[0].dataPointType, DataPointType.SUM);
+
+    assert.strictEqual(
+      metrics[0].descriptor.description,
+      'The number of connections that are currently in the state referenced by the attribute "state".'
```

Comment:
Could it be that this test is not being run? :thinking: 
The description has changed, so this should fail but [the job](https://github.com/open-telemetry/opentelemetry-js-contrib/actions/runs/4853420360/jobs/8649566069) somehow passes. :thinking: 
[TAV fails, though.](https://github.com/open-telemetry/opentelemetry-js-contrib/actions/runs/4853420381/jobs/8649578432?pr=1170)

Reply:
you're right. look at the [tav.yml](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/plugins/node/opentelemetry-instrumentation-mongodb/.tav.yml):
```
mongodb:
  jobs:
    - versions: ">=3.3 <4"
      commands: npm run test
    - versions: ">=4 <5"
      commands: npm run test-new-versions
``` 
 version 4 tests will run by the command 'test-new-versions' and version 3 tests will run by the command 'npm run test'. This is why you don't see the v4 test in the unit-test job. 

I'll fix the description issue so TAV will pass. 

But there is something still bothering me: in unit-test job it seems that v3 test didn't run at all:
![image](https://user-images.githubusercontent.com/85441461/235855189-f03430b6-b2c3-4c7d-99f2-7fc1a9128acb.png)
I'll take a look at this.

Reply:
Marking this as resolved as the tests are also not running on main. Thank you for creating the issue (#1487) :slightly_smiling_face: